### PR TITLE
samples: nrf_desktop: Remove recursive locking in buttons

### PR DESCRIPTION
**Pull request by pdunaj**
Jira:DESK-413

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

File: samples/nrf_desktop/src/hw_interface/buttons_matrix.c [hunk 1]
```diff
@@ -252,7 +272,7 @@ static void matrix_scan_fn(struct k_work *work)
 
 		case STATE_SUSPENDING:
 			state = STATE_ACTIVE;
-			err = suspend();
+			err = suspend_nolock();
 			if (!err) {
 				LOG_ERR("PDUNAJ Suspend me");
```
Comment (MarekPieta) [inline samples/nrf_desktop/src/hw_interface/buttons_matrix.c]: debug log?

Comment (pdunaj) [reply samples/nrf_desktop/src/hw_interface/buttons_matrix.c]: Thanks! removed

**Comment by pdunaj** [comment]
@pizi-nordic can you merge?

**Comment by pizi-nordic** [comment]
@pdunaj: Not before CI status.

**Comment by pdunaj** [comment]
@pizi-nordic it's green :)

**Pull request merged**